From 475208b518f6137e0797b4c18207a63953787a64 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Tue, 23 Dec 2025 14:08:29 +0800
Subject: [PATCH 254/254] mtk: hostapd: add csa/cca sanity check before disable
 bss

Add CSA/CCA sanity check before disable bss
Prohibit disabling BSS when CSA or CCA is in progress.
Otherwise, the CSA or CCA flag/parameter will not be cleared,
resulting in failure to re-enable BSS.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 hostapd/ctrl_iface.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 516eae7f0..35a79567c 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -1575,6 +1575,11 @@ static int hostapd_ctrl_iface_disable_bss(struct hostapd_data *hapd)
 	}
 #endif /* CONFIG_IEEE80211BE */
 
+	if (hapd->csa_in_progress || hapd->cca_in_progress) {
+		wpa_printf(MSG_ERROR, "Cannot disable BSS during CSA or CCA");
+		return -1;
+	}
+
 	if (hostapd_disable_bss(hapd) < 0) {
 		wpa_printf(MSG_ERROR, "Disabling of BSS failed");
 		return -1;
-- 
2.45.2

